src/main/java/duke/io/ReadFromFile.java

```diff
@@ -0,0 +1,85 @@
+package duke.io;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public abstract class ReadFromFile {
+    final static int LENGTH_BY = 4;
+    final static int LENGTH_AT = 4;
+
+    // File paths specified in an OS-independent way
+    static String root = System.getProperty("user.dir");
+    static java.nio.file.Path filePath = java.nio.file.Paths.get(root, "data", "out.txt");
+
+    public static void readFromMem(ArrayList<Task> tasks){
+        boolean fileExists = java.nio.file.Files.exists(filePath);
+        if (fileExists) {
+            int i = 0;
```

Comment:
Use a more descriptive name for your variable. Also, since this variable is only used in the `scanMem` method, it might make more sense to declare it there.